diff --git a/ckeditor.ckeditor.inc b/ckeditor.ckeditor.inc
index d69be0a..f6e4f4f 100644
--- a/ckeditor.ckeditor.inc
+++ b/ckeditor.ckeditor.inc
@@ -10,44 +10,6 @@ function ckeditor_ckeditor_plugin() {
   $plugin_dir_additional = ckeditor_plugins_path('local') . '/';
 
   /*
-   * External plugins
-   */
-  if (module_exists('ckeditor_swf') && file_exists(drupal_get_path('module', 'ckeditor_swf') . '/plugins/swf/plugin.js')) {
-    $plugins['ckeditor_swf'] = array(
-      'name' => 'swf',
-      'desc' => t('Support for the CKEditor SWF module'),
-      'path' => drupal_get_path('module', 'ckeditor_swf') . '/plugins/swf/',
-      'buttons' => FALSE,
-      'default' => 't'
-    );
-  }
-
-  if (module_exists('ckeditor_link') && file_exists(drupal_get_path('module', 'ckeditor_link') . '/plugins/link/plugin.js')) {
-    $plugins['ckeditor_link'] = array(
-      'name' => 'drupal_path',
-      'desc' => t('Support for the CKEditor Link module'),
-      'path' => drupal_get_path('module', 'ckeditor_link') . '/plugins/link/',
-      'buttons' => FALSE,
-      'default' => 't'
-    );
-  }
-
-  if (module_exists('linkit') && file_exists(drupal_get_path('module', 'linkit') . '/editors/ckeditor/plugin.js')) {
-    $plugins['linkit'] = array(
-      'name' => 'Linkit',
-      'desc' => t('Support for the Linkit module <em>(buttons: Linkit)</em>'),
-      'path' => drupal_get_path('module', 'linkit') . '/editors/ckeditor/',
-      'buttons' => array(
-        'Linkit' => array(
-          'label' => 'Linkit',
-          'icon' => 'linkit.png'
-        )
-      ),
-      'default' => 't'
-    );
-  }
-
-  /*
    * CKEditor build-in plugins
    */
   $_editor_path = ltrim($editor_path, './');
