--- superfish.module	2012-05-17 19:02:29.463861586 +0200
+++ superfish.module.new	2012-05-17 19:02:57.175860524 +0200
@@ -980,11 +980,11 @@
           if (is_array($settings['megamenu_exclude'])) {
-            $megamenu = (in_array($mlid, $settings['megamenu_exclude'])) ? 0 : $megamenu;
+            $megamenu_tmp = (in_array($mlid, $settings['megamenu_exclude'])) ? 0 : $megamenu;
           }
           else {
-            $megamenu = ($settings['megamenu_exclude'] == $mlid) ? 0 : $megamenu;
+            $megamenu_tmp = ($settings['megamenu_exclude'] == $mlid) ? 0 : $megamenu;
           }
           // Send the result to the sub-menu.
-          $sfsettings['megamenu'] = $megamenu;
+          $sfsettings['megamenu'] = $megamenu_tmp;
         }
-        if ($megamenu == 1) {
+        if ($megamenu_tmp == 1) {
           $megamenu_wrapper = ($menu_item['link']['depth'] == $settings['megamenu_depth']) ? TRUE : FALSE;
